numpy/core/tests/test_shape_base.py

```diff
@@ -554,6 +552,8 @@ def test_empty_lists(self):
         assert_raises_regex(ValueError, 'empty', np.block, [])
         assert_raises_regex(ValueError, 'empty', np.block, [[]])
         assert_raises_regex(ValueError, 'empty', np.block, [[1], []])
+        assert_raises_regex(ValueError, 'empty', np.block, [1, []])
+        assert_raises_regex(ValueError, 'empty', np.block, [[], 2])
```

Comment:
These used to raise the error "List depths are mismatched", which I think was more useful

Reply:
We can keep that refactoring out of this PR, but generally, since you already have a sanitation routine, I think it is best for the "working" routine to be guaranteed to "succeed".

Reply:
I agree that moving the checks out of the "working" routine makes sense, but if possible I'd like to preserve the error priority

Reply:
See if you like the rewrite. It gives the user the same information.

```
np.block([1, []])
```

```
ValueError: Lists cannot be empty. An empty list was found at index arrays[1]
```

Reply:
I'd still like the old error to take precedence. Empty lists are likely to be value-dependent, but mixed-depth lists are probably wrong all the time - I'd prefer that the latter is seen first.

A simple fix would be to propagate an extra boolean flag from `_block_check_depths_match` indicating if the list is empty. I can make a small PR for that if you want me to take the load off you a little here.

Reply:
I'm pretty tired right now to think about this. You can just push to my branch no? I won't force push anymore.

Reply:
I reverted the test. but not smart enough right now to fix the error generation.